includes/BasicObjects/Bar.js

```diff
@@ -0,0 +1,318 @@
+if (!RSECTION) {
+    throw new Error("This script is only for RSECTION, it creates RSection Bars.");
+}
+
+// Bug 46956: has_concerete_reinforcement bad API value name
+// Bar.MultiVariable function - cannot set bar.info_number_of_bars??
+
+/**
+ * Creates RSECTION Bar
+ * @class
+ * @constructor
+ * @param {Number} no               Number of Bar, can be undefined
+ * @param {Number} start_point_no   Start point number
+ * @param {Number} end_point_no     End point number
+ * @param {Number} material_no      Material number
+ * @param {Number} layer_no         Layer number
+ * @param {String} comment          Comment, can be undefined
+ * @param {Object} params           Parameters, can be undefined
+ */
+function Bar (no,
+    start_point_no,
+    end_point_no,
+    material_no,
+    layer_no,
```

Comment:
Add Layer object. User needs to create layer first. Then we can add bars to layer.